### PR TITLE
Add remote option to update_code_strategy to rename remote to repository

### PR DESCRIPTION
**Pull request by jhhazelaar**
Adds new option to update code strategy. Fix feature request #3464

### CODE REVIEW

File: recipe/deploy/update_code.php [hunk 1]
```diff
@@ -107,6 +108,14 @@
         cd('{{release_path}}');
         run("$git clone -l $bare .");
         run("$git checkout --force $target");
+    } else if (get('update_code_strategy') === 'remote') {
+        cd('{{release_path}}');
+        run("$git clone -l $bare .");
+        run("$git remote set-url origin $repository", ['env' => $env]);
```
Comment (antonmedv) [inline recipe/deploy/update_code.php:114]: clone and remote strategies seem really close to each other. Let's combine remote to clone.

Comment (jhhazelaar) [reply recipe/deploy/update_code.php]: Changed it, hopefully this is beter.

File: recipe/deploy/update_code.php [hunk 2]
```diff
@@ -100,12 +101,20 @@
 
     run("$git remote update 2>&1", ['env' => $env]);
 
+
     // Copy to release_path.
     if (get('update_code_strategy') === 'archive') {
         run("$git archive $targetWithDir | tar -x -f - -C {{release_path}} 2>&1");
-    } else if (get('update_code_strategy') === 'clone') {
+    } else if (get('update_code_strategy') === 'clone' || get('update_code_strategy') === 'remote') {
```
Comment (antonmedv) [inline recipe/deploy/update_code.php:108]: Let's just remove `remote` strategy and add

```php
run("$git remote set-url origin $repository", ['env' => $env]);

             $target = get('branch', 'HEAD');
```

to `clone` strategy.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: recipe/deploy/update_code.php (hunk 2 above)
Comment (jhhazelaar) [reply recipe/deploy/update_code.php]: Agree! Changed this.

**Pull request merged**